src/Raven.Server/Documents/Queries/GraphQueryRunner.IntermediateResults.cs

```diff
@@ -21,8 +21,19 @@ public void Add(Match match)
 
             public void Add(string alias, Match match, Document instance)
             {
-                //TODO: need to handle map/reduce results?
-                MatchesByAlias[alias][instance.Id] = match;
+                //we have map/reduce result that has no id since it is not a document
+                //in such case we have no choice but to use map/reduce result json as key since we have no other way to generate a key
+                //TODO : discuss with Oren & guys - perhaps there is a better way? (see http://issues.hibernatingrhinos.com/issue/RavenDB-12164)
```

Comment:
There is the reduce key, which isn't guaranteed to be unique (fan out)
Note that the `Data.ToString()` isn't unique either.